_scripts/customJS/tokenParser.js

```diff
@@ -0,0 +1,575 @@
+        const { AffiliationManager } = customJS
+
+        // Takes a token object and returns a formatted string
+
+        // reused from https://stackoverflow.com/questions/171251/how-can-i-merge-properties-of-two-javascript-objects-dynamically
+
+        function merge_options(obj1, obj2) {
+            var obj3 = {};
+            if (obj1) for (var attrname in obj1) { obj3[attrname] = obj1[attrname]; }
+            if (obj2) for (var attrname in obj2) { obj3[attrname] = obj2[attrname]; }
+            return obj3;
+        }
+
+        // overrides that are used elsewhere //
+        // overrides.dateInfo replaces the dateInfo object that is computed by DateManager //
+
+        let metadata = merge_options(file.frontmatter, overrides)
+
+        if (targetDate) targetDate = DateManager.normalizeDate(targetDate)
+        let pageDateInfo = metadata.dateInfo ?? DateManager.getPageDates(metadata, targetDate)
+        let displayDefaults = NameManager.getDisplayData(metadata)
+        let sourcePageType = NameManager.getPageType(metadata)
+
+        let value = "";
+
+
+        // set formatting options //  
+        // define four types of formatters //
+        // none: no formatting, just return the value //
+        // name: apply name formatting, expects a nameObject object //
+        // date: apply date formatting, expects a normalizeDate object //
+        // whereabout: build a location chain from this starting point //
+        // casing: just apply casing //
+
+        let formatter = "none"
+
+        switch (token.token) {
+            // date options - value is expected to be a normalizedDate object //
+            // to do: actually check this //
+            case "startdate":
+                if (pageDateInfo.startDate) value = pageDateInfo.startDate
+                formatter = "date"
+                break;
+            case "targetdate":
+                value = targetDate ? targetDate : DateManager.getTargetDateForPage(metadata)
+                formatter = "date"
+                break;
+            case "enddate":
+                if (pageDateInfo.endDate) value = pageDateInfo.endDate
+                formatter = "date"
+                break;
+            case "lastknowndate":
+                let wb = WhereaboutsManager.getWhereabouts(metadata, targetDate)
+                if (wb.lastKnown.location && wb.lastKnown.awayEnd.sort <= targetDate.sort) {
+                    value = wb.lastKnown.awayEnd
+                } else {
+                    if (targetDate) {
+                        value = targetDate
+                    } else {
+                        value = DateManager.getTargetDateForPage(metadata)
+                    }
+                }
+                formatter = "date"
+                break;
+            // end date options //
+
+            // start name options - can be a string or a name object //
+            case "name":
+                value = metadata.name ? metadata.name : file.name
```

Comment:
This will break files that have an actual name attribute in metadata that doesn’t match the filename. (Ie some of the Cleenseau people who non-ascii names). We should only use the name instead of the filename is the name is a normalized name object)

Reply:
Note, it only breaks them if the name is not also in the set of aliases. So I guess a pretty edge case, in that normally you'd expect the name, if set to something other than the filename, to be an alias. Not obvious if this is worth fixing.